src/components/ImageLightbox/component.js

```diff
@@ -0,0 +1,179 @@
+    const shareImage = onShare || this.shareImage;
+    const downloadImage = onDownload || this.downloadImage;
+    const openImage = onOpen || this.openImage;
+
+    return (currentIndex) => {
+      const currentImage = imageUrls[currentIndex];
+      const url = currentImage && currentImage.url;
+
+      // Check if we have a remote URL
+      const parsed = parseURL(url);
+      const isRemoteUrl = parsed && parsed.protocol && parsed.protocol.includes('http');
+
+      return (
+        <View style={styles.imageModalFooter}>
+          {/* Close */}
+          <TouchableOpacity style={styles.iconContainer} onPress={onClose}>
+            <Icon
+              style={[styles.icon, styles.closeIcon]}
+              name={Platform.select({ ios: 'ios-close', android: 'md-close' })}
+            />
+          </TouchableOpacity>
+
+          {/* Open */}
+          {isRemoteUrl &&
+            <TouchableOpacity
+              style={styles.iconContainer}
+              onPress={() => openImage(url)}
+            >
+              <Icon
+                style={[styles.icon, styles.openIcon]}
+                name={Platform.select({ ios: 'ios-open-outline', android: 'md-open' })}
+              />
+            </TouchableOpacity>
+          }
+
+          {/* Share */}
+          <TouchableOpacity
+            style={styles.iconContainer}
+            onPress={() => shareImage(url)}
+          >
+            <Icon
+              style={styles.icon}
+              name={Platform.select({ ios: 'ios-share-outline', android: 'md-share' })}
+            />
+          </TouchableOpacity>
+
+          {/* Download */}
+          {/*
+          Disabled for now ...
+          Several issues with this at the moment:
+            1. Need to find a way to download images to gallery on android.
+            2. `CameraRoll` has `saveToCameraRoll` which can save remote urls to iOS only.
+                see: https://facebook.github.io/react-native/docs/cameraroll.html#savetocameraroll
+            3. To get `CameraRoll.saveToCameraRoll` to work, you need to ask user for permission beforehand, otherwise app just crashes.
+          */}
+          {/* <TouchableOpacity
+            style={styles.iconContainer}
+            onPress={() => downloadImage(url)}
+          >
+            <Icon
+              style={styles.icon}
+              name={Platform.select({ ios: 'ios-cloud-download-outline', android: 'md-cloud-download' })}
+            />
+          </TouchableOpacity> */}
+        </View>
+      );
+    };
+  }
+
+  renderImage = (props) => {
+    const uri = (props && props.source && props.source.uri) || '';
+
+    /*
+    Data url images don't work on android with FastImage.
+    Thus we have to fallback to a regular image component.
+
+    Same thing is done in `PostImage`
+
+    Relevant PRs:
+      - https://github.com/DylanVann/react-native-fast-image/pull/91
+      - https://github.com/DylanVann/react-native-fast-image/pull/205
+    */
```

Comment:
Seems like both of these have been merged.

Reply:
Yup they have, just waiting on the author to release a new version. Once that is done i'll go ahead and update everything on another branch.